### ISSUE TITLE
function to generate dummy objects fails for classes with explicitly defined paths.

### ISSUE THREAD

**Issue by vddesai1871** [HTTP-APIs/hydrus#317]
### I'm submitting a 
- [x] bug report.
- [ ] feature request.

### Current Behaviour:

This function is used to create dummy classes for test purposes.
```
def gen_dummy_object(class_, doc):
    """Create a dummy object based on the definitions in the API Doc."""
    object_ = {
        "@type": class_
    }
    if class_ in doc.parsed_classes:
        for prop in doc.parsed_classes[class_]["class"].supportedProperty:
            if "vocab:" in prop.prop:
                prop_class = prop.prop.replace("vocab:", "")
                object_[prop.title] = gen_dummy_object(prop_class, doc)
            else:
                object_[prop.title] = ''.join(random.choice(
                    string.ascii_uppercase + string.digits) for _ in range(6))
        return object_
```
The first parameter of the function `class_` is ambiguous. Its not clear whether it should take class.title as its value or class.path as its value. In either case, it will fail for classes with explicitly given path names.
Suppose the class for which we want to create dummy objects has explicitly given path.
1) And suppose the `class_` parameter takes class.title as value and use that to create dummy object, 
then ` if class_ in doc.parsed_classes:` will return false. And hence the dummy object won't be created successfully.
2) now suppose the `class_` parameter takes class.path as value, 
then 
```
 object_ = {
        "@type": class_
    }
```
will create an invalid object as the `@type` property must have class.tilte as its value not class.path.
### Expected Behaviour:

From the parameter name, it should be clear that it takes class_path (or class_title, either will work) as value and then inside the function we can use it to retrieve class.title and set `@type` correctly, anything else won't need to be changed.
(Apologies for the long explanation, it was really an insidious bug ;))
### Steps to reproduce:

Create an apiDoc having class with explicitly defined path, and try to run tests or run `gen_dummy_obj()` function
### Do you want to work on this issue?

Yes

**Comment by Mec-iS** [comment]
I think `class_` is a HydraClass object not a `str`. I agree that parameters shall be documented and their types explicitly documented a well.

**Comment by vddesai1871** [comment]
>I think class_ is a HydraClass object not a str.

From function body and function calls it is clear we are expecting `class_` to be a string. But it was not clear whether it is `class.title` or `class.path`. We were passing `class.title` as value in function calls which is bound to fail when the class has explicitly defined path. I was suggesting to use class_title or class_path as the parameter and change the code accordingly. 
But yeah using HydraClass object can also work and it will simplify the code ( we will not need to pass `doc` as another parameter).

**Comment by xadahiya** [comment]
Fixed in #422
